src/averageAges.js

```diff
@@ -55,7 +81,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredTab = (onlyWithSon)
+    ? people.filter(filterMan).filter(filterChilds)
+    : people.filter(filterChilds);
+
+  function filterMan(person) {
```

Comment:
I see this function is the same as in **calculateMenAverageAge**. Instead of creating this function 2 times you can create it one time (in global scope) and use it inside where you need it. Check if you can extract some other helper functions that appears one than one time.
```suggestion
  function isMan(person) {
```